File: XYMenu/XYMenu/XYMenu/XYMenuItem.m
//
//  XYMenuItem.m
//  XYMenu
//
//  Created by FireHsia on 2018/1/26.
//  Copyright © 2018年 FireHsia. All rights reserved.
//

#import "XYMenuItem.h"


@interface XYMenuItem ()

@property (nonatomic, strong) UIImageView *iconImage; // 图标icon
@property (nonatomic, strong) UILabel *titleLab; // title
@property (nonatomic, copy) NSString *iconName;
@property (nonatomic, copy) NSString *title;

@end

@implementation XYMenuItem

- (instancetype)initWithIconName:(NSString *)iconName title:(NSString *)title
{
    if (self = [super init]) {
        self.backgroundColor = [UIColor clearColor];
        _iconName = iconName;
        _title = title;
    }
    return self;
}

- (void)setUpViewsWithRect:(CGRect)rect
{
    self.frame = rect;
    CGFloat kItemHeight = self.bounds.size.height;
    CGFloat iconHeight = kItemHeight / 3;
    [self addSubview:self.iconImage];
    [self addSubview:self.titleLab];
    self.iconImage.frame = CGRectMake(iconHeight, iconHeight, iconHeight, iconHeight);
    CGFloat iconMaxY = CGRectGetMaxY(self.iconImage.frame);
    self.titleLab.frame = CGRectMake(iconMaxY + (iconHeight * 3 / 4), iconHeight, iconHeight * 3, iconHeight);
}

- (UIImageView *)iconImage
{
    if (!_iconImage) {
        _iconImage = [[UIImageView alloc] initWithImage:[UIImage imageNamed:_iconName]];
    }
    return _iconImage;
}

- (UILabel *)titleLab
{
    if (!_titleLab) {
        _titleLab = [[UILabel alloc] init];
        _titleLab.textColor = [UIColor whiteColor];
        _titleLab.text = _title;
        _titleLab.font = [UIFont systemFontOfSize:16.0];
        _titleLab.backgroundColor = [UIColor clearColor];
    }
    return _titleLab;
}

@end

File: XYMenu/XYMenuSwift/XYMenuItem.swift
//
//  XYMenuItem.swift
//  XYMenuSwiftDemo
//
//  Created by FireHsia on 2018/2/2.
//  Copyright © 2018年 FireHsia. All rights reserved.
//

import UIKit

class XYMenuItem: UIView {
    
    var icon: String
    var title: String
    
    // MARK: LazyLoad
    lazy var iconImage: UIImageView = {
        var image = UIImageView.init(image: UIImage.init(named: icon))
        return image
    }()
    
    lazy var titleLab: UILabel = {
        var label = UILabel.init(frame: CGRect.zero)
        label.text = title
        label.textColor = .white
        label.font = UIFont.systemFont(ofSize: 16.0)
        label.backgroundColor = .clear
        return label
    }()
    
    // MARK: 初始化方法
    init(_ iconStr: String,_ titleStr: String) {
        icon = iconStr
        title = titleStr
        super.init(frame: CGRect.zero)
    }
    
    func setUpViews(_ rect: CGRect) {
        frame = rect
        let kItemHeight = bounds.size.height
        let iconHeight = kItemHeight / 3
        addSubview(iconImage)
        addSubview(titleLab)
        iconImage.frame = CGRect(x: iconHeight, y: iconHeight, width: iconHeight, height: iconHeight)
        let iconMaxY = iconImage.frame.maxY
        titleLab.frame = CGRect(x: iconMaxY + (iconHeight * 3) / 4, y: iconHeight, width: iconHeight * 3, height: iconHeight)
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
}
